### PR TITLE
DES-1676 - Add ability to update project information in manage panel..

### PR DESCRIPTION
**Pull request by duckonomy**
## Overview: ##
Add ability to update project information (name and description) in the Manage panel.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-1676](https://jira.tacc.utexas.edu/browse/DES-1676)

## Summary of Changes: ##
Changed `users-panel.component.ts` to implement UI for changing project data.
Changed `projects.service.ts` to make additional calls to the backend to update project data.

## Testing Steps: ##
1. Go to the manage panel in the hazmapper app.
2. Double click or press the edit icon on either name or description.
3. When done, click out of the input or click the check icon.
4. Confirm that the UI is updated.
5. Refresh the browser.
6. Go back to the manage panel.
7. Confirm that the change persisted.

## UI Photos:
Screenshot from 2020-10-23 12-52-20
Screenshot from 2020-10-23 12-52-20
Screenshot from 2020-10-23 12-39-48

## Notes: ##
Needs some refactoring for duplicate functionality. And more brushing up on UI.
Depends on TACC-Cloud/geoapi#39

### CODE REVIEW

File: src/app/components/users-panel/users-panel.component.html [hunk 1]
```diff
@@ -15,9 +15,55 @@ <h4> Current Members </h4>
...
+        <i class="fas fa-check" style="margin: auto" (click)="hideNameInput()" *ngIf="nameInputShown"></i>
+      </div>
+      <div class="display-error" *ngIf="nameInputError">
+        {{nameErrorMessage}}
+      </div>
+
       <h6>Description:</h6>
-      <p>{{activeProject.description}}</p>
+      <div style="display: flex; justify-content: space-between">
+        <div style="display: inline-block; width: 100%">
```
Comment (nathanfranklin) [inline src/app/components/users-panel/users-panel.component.html:46]: I'm also curious if a new component should be created to wrap up the edit input.  Not certain if that is overkill but used 2 places here and could be used other places going forward.

**Review by nathanfranklin — COMMENTED** [review]
Looks and works great 👍 

I left some comments.

File: src/app/components/users-panel/users-panel.component.html (hunk 1 above)
Comment (duckonomy) [reply src/app/components/users-panel/users-panel.component.html:46]: @nathanfranklin I'll make a ticket for that.

Comment (nathanfranklin) [reply src/app/components/users-panel/users-panel.component.html:46]: > @nathanfranklin I'll make a ticket for that.

perfect!

File: src/app/components/users-panel/users-panel.component.html [hunk 2]
```diff
@@ -15,9 +15,55 @@ <h4> Current Members </h4>
     <div class="">
       <h4> Map </h4>
       <h6>Name:</h6>
```
[the remark this answers is not in the document]
Comment (duckonomy) [reply src/app/components/users-panel/users-panel.component.html:17]: @nathanfranklin Do you mean the header or content? I definitely think that there should be some visual delineation between the header and content of each sections. But I'm not sure what the best way would be to differentiate between the sections.
I'll try different things out.

[the remark this answers is not in the document]
Comment (nathanfranklin) [reply src/app/components/users-panel/users-panel.component.html:17]: the header and content.  Like how "Description:" has the same font as "cool" in the screenshot.  I don't have any suggestions besides making this bigger or increasing the weight.  but not certain if that would be an improvement so maybe its good already.

**Review by nathanfranklin — APPROVED** [review]
Looks great!  💯  the new component is nice!

**Pull request merged**